### PR TITLE
Additional Configs

### PR DESCRIPTION
**Pull request by EPuncker**
- Added !buyhouse level and premium requirement config (#2879 )
- Added setting to enable/disable change sex and buy premium command
- Moved a few talkaction settings config.lua file

### CODE REVIEW

**Comment by MillhioreBT** [comment]
I don't like the idea of ​​the `script config.lua` file but I do like the reason.

**Comment by DSpeichert** [comment]
The second config is quite confusing.
Can't we just add additional options to the "main" config.lua that the C++ code will simply ignore?

**Comment by soul4soul** [comment]
I'm don't like the idea of combining all of the scripts with a common config file. I do like some of the enhancements made to the individual scripts and would like to see the specific configs stay in those files. EG. keep the new config added for dice on depot in the dice.lua file.

**Comment by EPuncker** [comment]
> The second config is quite confusing.
> Can't we just add additional options to the "main" config.lua that the C++ code will simply ignore?

I didn't knew it was possible 😄 gonna do it

**Comment by DSpeichert** [comment]
So config.lua is implicitly available throughout Lua scope? I don't see the libs/ executing it anywhere.

**Comment by EPuncker** [comment]
> So config.lua is implicitly available throughout Lua scope? I don't see the libs/ executing it anywhere.

yeah its kinda weird, there are some that works and some that don't, will investigate further

File: config.lua.dist [hunk 1]
```diff
@@ -92,6 +100,12 @@ yellMinimumLevel = 2
 yellAlwaysAllowPremium = false
 forceMonsterTypesOnLoad = true
 cleanProtectionZones = false
+changeSexEnabled = true
+changeSexDaysCost = 3
+premiumToOfflineTraining = true
+deathListEnabled = true
+maxDeathRecords = 5
+useWormsToFishing = true
```
Comment (DSpeichert) [inline config.lua.dist:108]:
```suggestion
useWormsForFishing = true
```

**Review by DSpeichert — CHANGES_REQUESTED** [review]

**Comment by nekiro** [comment]
Hoenstly I find these redundant and config bloating, why creating config values for easily editable lua scripts?

**Comment by EPuncker** [comment]
> Honestly I find these redundant and config bloating, why creating config values for easily editable lua scripts?
> Some of them like death related are okay, but "useWormsForFishing" is kind of redundant to me.

I find it better to have all of them in a single place, but yes that is kinda of a matter of opinion

**Comment by peonso** [comment]
Bloating. Config should only deal with server setup (map name/ip/database) and hardcoded stuff. If you add everysingle script option to config.lua because someone want it there we can reach thousand lines really fast.

**Comment by DSpeichert** [comment]
I have reconsidered my earlier opinion and decided that config.lua is not the place to gather all config options from the datapack.
IMHO it would be clearer to the user when config.lua focuses of the built-in engine features, which generally means fewer and fewer things as things move to Lua.

What we need instead is a good datapack README, showing which scripts are responsible for which behavior, so that a user can go and reconfigure them.

I'm primarily thinking about TFS as an engine that can be used with the widest range of datapacks, therefore comingling the core config and datapack config is confusing.

**Pull request closed without merging**